azure-keyvault/azure/keyvault/secrets/_client.py

```diff
@@ -52,43 +63,40 @@ def __init__(self, credentials, vault_url, config=None, **kwargs):
         if not vault_url:
             raise ValueError('vault_url')
 
-        self.vault_url = vault_url
+        self._vault_url = vault_url
         config = config or SecretClient.create_config(**kwargs)
         transport = RequestsTransport(config.connection)
         policies = [
             config.user_agent,
             config.headers,
-            BearerTokenCredentialPolicy(credentials),
+            _BearerTokenCredentialPolicy(credentials),
             config.redirect,
             config.retry,
             config.logging,
         ]
-        client_models = {
-            'Secret': Secret,
-            'SecretAttributes': SecretAttributes,
-            '_BackupResult': _BackupResult
-        }
-        self._serialize = Serializer(client_models)
-        self._deserialize = Deserializer(client_models)
         self._pipeline = Pipeline(transport, policies=policies)
 
-    def get_secret(self, name, version=None, **kwargs):
+    @property
+    def vault_url(self):
+        return self._vault_url
+
+    def get_secret(self, name, version, **kwargs):
```

Comment:
specifying the version is mandatory now?

Reply:
Short answer: yes.

Longer answer: `version` no longer has a default value here, so a caller must provide one, but the Key Vault API hasn't changed, i.e. one could pass the empty string for `version` and get the old behavior (shhh don't tell anyone 😶 ).